src/frontends/tensorflow/src/saved_model.hpp

```diff
@@ -0,0 +1,240 @@
+std::basic_string<wchar_t> get_saved_model_name<wchar_t>();
+template <>
+std::basic_string<wchar_t> get_variables_index_name<wchar_t>();
+#endif
+
+// Stores information about variables index
+class SavedModelVariablesIndex {
+    // Contains maximum amount of shards, used for creating corrext extension
+    int32_t m_total_shards;
+    // Contains BundleEntryProto variables list, readed from .index file
+    std::map<std::string, std::vector<char>> m_variables_index;
+    // List of opened data files for using with BundleEntryProto
+    std::map<int32_t, std::shared_ptr<std::ifstream>> m_data_files;
+    // List of mapped variables which could be read using TrackableObjectGraph
+    std::map<std::string, std::string> m_variables_map;
+
+public:
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::string& path);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    /// \brief Reads variables from opened variable index file. Can cause an asserts in case of issues.
+    /// \param vi_stream Opened stream file, file pointer doesn't matter, it will be rewind internally.
+    /// \param path A path to file with variables data
+    /// \returns Returns true in case of everything loads successfully, false otherwise
+    bool read_variables(std::ifstream& vi_stream, const std::wstring& path);
+#endif
+
+    /// \brief Returns data and size of data of stored variable
+    /// \param name Name of variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_variable(const std::string& name, char** data, size_t* size) {
+        auto varItem = m_variables_index.find(name);
+        if (varItem == m_variables_index.end()) {
+            return false;
+        }
+        if (data != nullptr) {
+            *data = varItem->second.data();
+        }
+        if (size != nullptr) {
+            *size = varItem->second.size();
+        }
+        return true;
+    }
+
+    /// \brief Returns data and size of data of mapped variable from trackable object graph to variables index
+    /// \param name Name of a mapping variable
+    /// \param data Pointer on a pointer where data pointer will be returned
+    /// \param size Pointer on a variable which will stores data size
+    /// \returns Returns true in case variable was found, false otherwise (data and size will be untouched)
+    bool get_mapped_variable(const std::string& name, char** data, size_t* size) {
```

Comment:
can we use `std::vector<char>` instead of `char** data, size_t* size`?

Reply:
Using current implementation I don't recommend, because get_variable uses it as a flags for memory allocation.

You can call get_mapped_variable("name", nullptr, &entry_size) and get only size, instead of returning data.

Moreover, currently it returns pre-allocated memory without copying. If you pass a vector here - it will add memory copy from existing read-only memory.

Reply:
I propose to use `shared_ptr<char[]>& ` instead because now you manage memory not carefully. Who can guarantee that this memory will be released in case of trouble-shooting or much earlier than this is needed.
I am against this approach.